src/native/libs/System.Security.Cryptography.Native/openssl.c

```diff
@@ -168,6 +176,8 @@ The encoded value of the version, otherwise:
 */
 int32_t CryptoNative_GetX509Version(X509* x509)
 {
+    // No error queue impact.
```

Comment:
I... think this is true enough? There are errors in the error path here, but that depends on how far you can get with an X509 certificate that contains a bogus version. I only dug in to this one because it's not a `get0`.

1. `X509_get_version` [calls](https://github.com/openssl/openssl/blob/31b7f23d2f958491d46c8a8e61c2b77b1b546f3e/crypto/x509/x509_set.c#L113) `ASN1_INTEGER_get`.
2. `ASN1_INTEGER_get` [calls](https://github.com/openssl/openssl/blob/1c0eede9827b0962f1d752fa4ab5d436fa039da4/crypto/asn1/a_int.c#L556) `ASN1_INTEGER_get_int64`
3. `ASN1_INTEGER_get_int64` [calls](https://github.com/openssl/openssl/blob/1c0eede9827b0962f1d752fa4ab5d436fa039da4/crypto/asn1/a_int.c#L527) `asn1_string_get_int64`.
4. We start erroring here for bad inputs.

Reply:
Hm.  I think I made an assumption that during d2i_X509 something would have already checked that the version wasn't unknown (anything other than 0, 1, or 2).  Assuming that has already been checked for an X509 value then none of the error conditions in that call chain can be met.

There's also a secondary interpretation of that comment, which is "since we can't report failure from this function, we don't throw an OpenSslException".  And I just checked, the caller throws a bare CryptographicException if the answer comes back negative, otherwise it just trusts the API.